src/commands/NewCamelRouteCommand.ts

```diff
@@ -59,6 +60,8 @@ export class NewCamelRouteCommand {
 				return { language: 'Yaml', extension: 'camel.yaml', placeHolder: 'sample-route' };
 			case 'JAVA':
 				return { language: 'Java', extension: 'java', placeHolder: 'SampleRoute' };
+			case 'XML':
+				return { language: 'Xml', extension: 'xml', placeHolder: 'sample-route' };
```

Comment:
TODO:

extension should be `.camel.xml` but currently it is not possible. I will open issue at upstream to fix this in next camel version